### ISSUE TITLE
undefined errors happening commonly as of late

### ISSUE THREAD

**Issue by esotericist** [nornagon/cdda-guide#125]
was trying to check a thing related to plastic molds, got this result:
![image](https://user-images.githubusercontent.com/1569754/224791195-389501d1-4274-4bd3-a344-98ed29d4b5df.png)
i've checked 0.G, 0.F, at least experimental build a day for the last two weeks, and a bunch in scattered months going all the way to the oldest listed (july 27, 2022).

i arrived on that specific url after clicking on the link in this search:
![image](https://user-images.githubusercontent.com/1569754/224791371-0f427e67-43a7-4bfe-9eee-69480354ac48.png)

i've experienced this on a number of other items recently and generally got it to work by shuffling versions to a few days (to in one case a couple weeks) back, so i thought it was just a transient problem with something in recent builds, but this appears to be an example of an item that straight up is broken all the way.

browser is chrome Version 110.0.5481.181, in case that somehow matters.

**Comment by esotericist** [comment]
... actually i think it's affecting everything on every build, now, so it's worse today than last night. hopefully the very recent commits i see are you being on that.

**Comment by nornagon** [comment]
Thanks for the heads up, should be fixed soon!

**Comment by nornagon** [comment]
the background here is that https://github.com/CleverRaven/Cataclysm-DDA/pull/62180 added some requirements that had no components list, which the Guide barfed on due to an incorrect schema. Then I discovered that TypeScript _should_ have alerted me to the issue but didn't, because I hadn't enabled strict null checking. So I enabled that and it found a whole bunch of other (mostly, but not entirely) spurious issues around the codebase. One of them was in the error-checking module, which is supposed to capture errors and display a nice message instead of an empty page, but I fixed it wrong 😅. Normally the tests catch this sort of thing, but ironically, the error-catching module is not tested. ha ha ha ha.
